pythia/utils/text_utils.py

```diff
@@ -289,3 +289,91 @@ def best_score(self):
             i = self.complete_seqs_scores.index(max(self.complete_seqs_scores))
             captions = torch.FloatTensor(self.complete_seqs[i]).unsqueeze(0)
         return captions
+
+
+class NucleusSampling:
+    def __init__(self, vocab, threshold=0.8):
+        self.vocab = vocab
+        self.vocab_size = vocab.get_size()
+        # Threshold for sum of probability
+        self.threshold = threshold
+        # Lists to store completed sequence
+        self.complete_seq = []
+
+    def init_batch(self, sample_list):
+        setattr(
+            self,
+            "seq",
+            sample_list.answers.new_full(
+                (1, 1), self.vocab.SOS_INDEX, dtype=torch.long
+            ),
+        )
+        setattr(
```

Comment:
what's the reason for initializing this here when you are replacing it in line 330
https://github.com/facebookresearch/pythia/blob/5370d1ba67958980dd3ee2d54564857720cfa37c/pythia/utils/text_utils.py#L330

Reply:
Right. Thinking of it now, top_m_scores isn't required to be defined as an attribute. Will fix.